Change Basic Iterative Method (BIM)

beat-buesser opened the pull request:
# Description

The current implementation of the Basic Iterative Method asks for a value of eps but does not project to eps.

After discussing with @mathsinn we think that BIM should be implemented as a special case of PGD using no random initialisation (num_random_init=0) and norm=np.inf.

Fixes #90 

## Type of change

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

mathsinn commented on art/attacks/iterative_method.py:
```diff
@@ -48,99 +46,14 @@ def __init__(self, classifier, norm=np.inf, eps=.3, eps_step=0.1, max_iter=100,
         :type eps: `float`
         :param eps_step: Attack step size (input variation) at each iteration.
         :type eps_step: `float`
-        :param max_iter: The maximum number of iterations.
```
I think this parameter should not be removed - can you please double check?

mathsinn replied:
On the other hand, I think the parameter `norm` should be removed.

beat-buesser replied:
I have resolved this with a38b34d8b5cfcbec46cee3b97fb6b9c98ebae283

mathsinn commented on art/attacks/iterative_method.py:
```diff
@@ -48,99 +46,14 @@ def __init__(self, classifier, norm=np.inf, eps=.3, eps_step=0.1, max_iter=100,
         :type targeted: `bool`
-        :param num_random_init: Number of random initialisations within the epsilon ball. For num_random_init=0
-            starting at the original input.
+        :param num_random_init: Number of random initialisations within the epsilon ball. For random_init=0 starting at
```
I think the `num_random_init` parameter should be removed.

beat-buesser replied:
I have resolved this with I have resolved this with a38b34d8b5cfcbec46cee3b97fb6b9c98ebae283

beat-buesser commented:
@mathsinn Thank you for your review. During addressing your review requests I have noticed that the methods `set_params` also need to be updated and consistently applied in the `__init__` methods of `ProjectedGradientDescent` and `FastGradientMethod`
